### DISCUSSION TITLE
npm ERR! code ERESOLVE for npm install react-hook-form

### DISCUSSION THREAD

**Discussion by aaabhijith13** [react-hook-form/react-hook-form#8604] [Q&A]
npm install react-hook-from gives me the following issue, I am unclear on how to fix it: 

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @material-ui/core@4.12.4
npm ERR! Found: @types/react@18.0.11
npm ERR! node_modules/@types/react
npm ERR!   peerOptional @types/react@"*" from @material-ui/types@5.1.0
npm ERR!   node_modules/@material-ui/types
npm ERR!     @material-ui/types@"5.1.0" from @material-ui/core@4.12.4
npm ERR!     node_modules/@material-ui/core
npm ERR!       peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
npm ERR!       node_modules/@material-ui/icons
npm ERR!         @material-ui/icons@"^4.11.3" from the root project
npm ERR!       1 more (the root project)
npm ERR!     @material-ui/types@"5.1.0" from @material-ui/styles@4.11.5
npm ERR!     node_modules/@material-ui/styles
npm ERR!       @material-ui/styles@"^4.11.5" from @material-ui/core@4.12.4
npm ERR!       node_modules/@material-ui/core
npm ERR!         peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
npm ERR!         node_modules/@material-ui/icons
npm ERR!         1 more (the root project)
npm ERR!   peerOptional @types/react@"^17.0.0 || ^18.0.0" from @mui/base@5.0.0-alpha.84
npm ERR!   node_modules/@mui/base
npm ERR!     @mui/base@"5.0.0-alpha.84" from @mui/material@5.8.3
npm ERR!     node_modules/@mui/material
npm ERR!       peer @mui/material@">=5.0.0" from @devexpress/dx-react-grid-material-ui@3.0.3
npm ERR!       node_modules/@devexpress/dx-react-grid-material-ui
npm ERR!         @devexpress/dx-react-grid-material-ui@"^3.0.3" from the root project
npm ERR!       5 more (@mui/icons-material, @mui/lab, @mui/x-data-grid, ...)
npm ERR!   13 more (@mui/icons-material, @mui/lab, @mui/base, @mui/material, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional @types/react@"^16.8.6 || ^17.0.0" from @material-ui/core@4.12.4
npm ERR! node_modules/@material-ui/core
npm ERR!   peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
npm ERR!   node_modules/@material-ui/icons
npm ERR!     @material-ui/icons@"^4.11.3" from the root project
npm ERR!   @material-ui/core@"^4.12.4" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: @types/react@17.0.47
npm ERR! node_modules/@types/react
npm ERR!   peerOptional @types/react@"^16.8.6 || ^17.0.0" from @material-ui/core@4.12.4
npm ERR!   node_modules/@material-ui/core
npm ERR!     peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
npm ERR!     node_modules/@material-ui/icons
npm ERR!       @material-ui/icons@"^4.11.3" from the root project
npm ERR!     @material-ui/core@"^4.12.4" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

**Comment by bluebill1049** [comment]
> npm ERR! Could not resolve dependency:
> npm ERR! peerOptional @types/react@"^16.8.6 || ^17.0.0" from @material-ui/core@4.12.4

That's the issue, please check other deps.
